src/main/java/org/openstreetmap/atlas/checks/validation/intersections/LineCrossingWaterBodyCheck.java

```diff
@@ -66,6 +72,75 @@
+            .forDefinition(OFFENDING_LINEITEMS);
+    // Whether we should flag buildings that cross waterbodies
+    private static final Predicate<Edge> HIGHWAY_TYPE_TO_FLAG = object -> object.highwayTag()
+            .isMoreImportantThanOrEqualTo(HighwayTag.PATH)
+            && Validators.isNotOfType(object, HighwayTag.class, HighwayTag.BUS_GUIDEWAY);
+    // Whether we should limit flags to only streets, buildings, and railways
+    private static final boolean DEFAULT_OFFENDING_LINEITEM_FILTER = false;
+    private boolean filterOffendingLineItems;
+    // Assume the object is an area based on atlas call
+    private static final Predicate<AtlasObject> IS_BUILDING = object -> Validators
+            .isNotOfType(object, BuildingTag.class, BuildingTag.NO);
+    private static final String BUILDING_TAGS_DO_NOT_FLAG = "public_transport->station,aerialway=station";
+    private static final TaggableFilter NONOFFENDING_BUILDINGS = TaggableFilter
+            .forDefinition(BUILDING_TAGS_DO_NOT_FLAG);
+
+    private static final String WATER_BODY_TAGS =
+            // Lakes
+            "natural->spring,hot_spring&name->*" + "|natural->lake,pond" + "|water:type->lake"
+                    + "|landuse->pond" + "|water->lake,pond,oxbow,salt_lake" +
+
+                    // Rivers
+                    "|natural->stream"
+                    + "|water->canal,river,lock,moat,riverbank,creek,stream,stream_pool"
+                    + "|waterway->river,riverbank,brook,ditch,stream,creek,canal,derelict_canal"
+                    + "|stream->*" + "|waterway->drain&name->*" + "|water->drain&name->*" +
+
+                    // Reservoirs
+                    "|water->reservoir" + "|water->dam&natural->water" + "|landuse->reservoir"
+                    + "|natural->reservoir" + "|seamark:type->dam&natural->water" +
+
+                    // Miscellaneous
+                    "|natural->water" + "|waterway->water" + "|water->water,perennial"
+                    + "|landuse->water" +
+
+                    // Wetlands
+                    "|wetland->tidalflat,reedbed" + "|water->tidalflat,reedbed"
+                    + "|natural->tidalflat,reedbed" +
+
+                    // Lagoons
+                    "|natural->lagoon" + "|water->lagoon" + "|waterway->lagoon" +
+
+                    // Intermittent/Dry lakes
+                    "|intermittent->dry" + "|seasonal->dry" + "|natural->dry_lake" +
+
+                    // Unique
+                    "|waterway->billabong,navigablechannel,river;stream,reservoir";
+    private static final TaggableFilter VALID_WATER_BODY_TAGS = TaggableFilter
+            .forDefinition(WATER_BODY_TAGS);
+
+    private static final String WATER_BODY_EXCLUDE_TAGS = "natural->dock,water_point,floodway,spillway,wastewater,waterhole"
+            + "|waterway->lock_gate,dock,water_point,floodway,spillway,wastewater,waterhole,culvert,dam,waterfall,fish_pass,dry_dock,construction,boat_lift,weir,breakwater,boatyard"
```

Comment:
Wondering if we can move these to json file and do validation here will be good. It's just my opinion. Please use your better judgement.